joblib/parallel.py

```diff
@@ -204,27 +173,24 @@ def __call__(self, out):
         self.parallel.n_completed_tasks += self.batch_size
         this_batch_duration = time.time() - self.dispatch_timestamp
 
-        if (self.parallel.batch_size == 'auto'
-                and self.batch_size == self.parallel._effective_batch_size):
-            # Update the smoothed streaming estimate of the duration of a batch
-            # from dispatch to completion
-            old_duration = self.parallel._smoothed_batch_duration
-            if old_duration == 0:
-                # First record of duration for this batch size after the last
-                # reset.
-                new_duration = this_batch_duration
-            else:
-                # Update the exponentially weighted average of the duration of
-                # batch for the current effective size.
-                new_duration = 0.8 * old_duration + 0.2 * this_batch_duration
-            self.parallel._smoothed_batch_duration = new_duration
-
+        self.parallel._pool.batch_completed(self.batch_size, this_batch_duration)
         self.parallel.print_progress()
         if self.parallel._original_iterator is not None:
             self.parallel.dispatch_next()
 
 
 ###############################################################################
+def set_parallel_backend(name, cls):
```

Comment:
Maybe this function should be called 'register_parallel_backend' to better reflect what is does. I think it would be worth adding some checks on the class registered (just to be sure it's a valid parallel backend).